### ISSUE TITLE
feat(py): Implement `VideoDecodingTransformer` for stateful video decoding in ML pipelines

### ISSUE THREAD

**Issue by fdicorato** [mosaico-labs/mosaico#479]
Currently, extracting compressed video streams (like H.264 or HEVC) for machine learning pipelines presents a critical architectural conflict. Stateful video formats rely on inter-frame compression (I-frames followed by P/B delta-frames). 

If a user extracts a sequence containing compressed video, he cannot apply temporal synchronization (`SyncTransformer` with `SyncHold`), or batch/shuffle the data for PyTorch or TensorFlow, because of the delta-frames. He must manually use the stateful decoder first, to correctly obtain the decoded image, and then process it via the `SyncTransformer` or any trainign pipeline. 

We want to automate this mechanism, via the definition of a decoding transformer.

### Proposed Solution
We introduce a `VideoDecodingTransformer` in the `mosaicolabs.ml` module. This transformer will sit *between* the `DataFrameExtractor` and the `SyncTransformer`. It will iterate through chronological sparse chunks, safely decode the byte streams into `PIL.Image` objects using a persistent session, and replace the raw byte columns with the reconstructed images. 

**Key Implementation Requirements:**
1. **Scikit-Learn Compatibility:** The class must implement `fit` and `transform`.
2. **Soft Dependency:** We do not want to force a `scikit-learn` dependency on core users. We must implement a `try...except ImportError` block in `mosaicolabs/ml/__init__.py` to provide dummy `BaseEstimator` and `TransformerMixin` fallback classes if `sklearn` is not installed.
3. **Format Routing:** The `transform` method must check the `ImageFormat` of the column. Stateful formats (H.264, HEVC) must be routed to `StatefulDecodingSession`, while stateless formats (JPEG, PNG) should fall back to `_StatelessDefaultCodec`.
4. **Memory Management:** The transformer must overwrite the raw `compressed_image.data` field with the decoded PIL.Image data and remove the `compressed_image.format` column after decoding.

### **Example Usage**
```python
from sklearn.pipeline import Pipeline
from mosaicolabs.ml import DataFrameExtractor, SyncTransformer, VideoDecodingTransformer

# 1. Define pipeline
pipeline = Pipeline([
    ('decoder', VideoDecodingTransformer(topics=["/front/camera/image"])),
    ('sync', SyncTransformer(target_fps=30.0))
])

# 2. Extract and transform
for sparse_chunk in extractor.to_pandas_chunks(topics=["/front/camera/image"]):
    # Safely decodes chronologically, THEN synchronizes
    dense_chunk = pipeline.transform(sparse_chunk)
```

**Feature to be implemented by branching from tag [mosaico-py/v0.4.1](https://github.com/mosaico-labs/mosaico/releases/tag/mosaico-py%2Fv0.4.1). The feature will be added to the new SDK release (0.4.2)**

### **Tasks / Acceptance Criteria**
- [x] Add the soft-dependency handling logic to `mosaicolabs/ml/__init__.py` and expose `__all__ = ["BaseEstimator", "TransformerMixin"]`.
- [x] Implement `VideoDecodingTransformer` in `mosaicolabs/ml/decoding_transformer.py`.
- [x] Add robust docstrings detailing the stateful nature of the transformer.
- [x] Ensure the `reset()` method correctly closes the C-level decoder resources.

**Comment by fdicorato** [comment]
working on this

**Comment by fdicorato** [comment]
solved here 29625c49990aa6f98adacfee58489c1d52d58d18
